### DISCUSSION TITLE
FSK demodulation

### DISCUSSION THREAD

**Discussion by Sumya99** [jgromes/RadioLib#472] [Q&A] [answered]
Hello sir,

I am using RFM95 module in FSK mode, which is sending "Hello world!" string. I want to demodulate the FSK signal using SDR dongle. Please let me know in what kind of ways I can get it?  

Kind regards

**Comment by jgromes** [comment, marked as answer]
I would suggest trying Universal Radio Hacker: https://github.com/jopohl/urh

It's very easy to setup and can decode FSK packets pretty much out-of -the-box.

**Reply by Sumya99** [reply]
Thank you for the replies,

First of all, I checked that RFM95 transmitter sends "Hello World!", RFM95 receiver receives it normally. When I use Universal Radio Hacker the result is not showing "Hello World!". What protocol is used in FSK_modem example?  Please help me I do not know what I did wrong.      

![image](https://user-images.githubusercontent.com/48245515/154610601-146138f6-27ee-467a-bdf5-358fd49226a9.png)

**Reply by jgromes** [reply]
Try switching signal view to demodulated, that will show you the 1s and 0s and will be helpful to determine if you need to do some filtering on the signal.

You also seem to have recorded a lot of the packets. That's probably not necessary, 2 - 3 will suffice for analysis. You can actually see a part of the packet in the ASCII view: "rld". Since it's cut, I would say that your samples/symbol is slightly off, or there's some noise being interpreted as a part of the preamble.

The packet format is very simple (it's shown in the RFM9x/SX127x datasheet). There's a pramble of alternating 0s and 1s, then a sync word (0x12 0xAD), then the length of the packet (0x0C), the payload "Hello World!", and then the 2-byte CRC field.
